tools/tck-regression/src/main/java/org/opencypher/tools/tck/regression/TckRegressionModel.java

```diff
@@ -0,0 +1,233 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Attribution Notice under the terms of the Apache License 2.0
+ *
+ * This work was created by the collective efforts of the openCypher community.
+ * Without limiting the terms of Section 6, any Derivative Work that is not
+ * approved by the public consensus process of the openCypher Implementers Group
+ * should not be described as “Cypher” (and Cypher® is a registered trademark of
+ * Neo4j Inc.) or as "openCypher". Extensions by implementers or prototypes or
+ * proposals for change that have been documented or implemented should only be
+ * described as "implementation extensions to Cypher" or as "proposed changes to
+ * Cypher that are not yet approved by the openCypher community".
+ */
+package org.opencypher.tools.tck.regression;
+
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableList;
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+import static org.opencypher.tools.tck.regression.TckResultsComparator.checkState;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TckRegressionModel {
+    static class Scenarios {
+        private final List<Scenario> all;
+        private final List<Scenario> passing;
+        private final List<Scenario> failed;
+
+        private Scenarios(List<Scenario> all, List<Scenario> passing, List<Scenario> failed) {
+            this.all = all;
+            this.passing = passing;
+            this.failed = failed;
+        }
+
+        static Scenarios create(TestSuite feature) {
+            List<Scenario> all = unmodifiableList(feature.testcase);
+
+            List<Scenario> passed = all.stream()
+                .filter(Scenario::isPassed)
+                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
+
+            List<Scenario> failed = all.stream()
+                .filter(Scenario::isFailed)
+                .collect(collectingAndThen(toList(), Collections::unmodifiableList));
+
+            return new Scenarios(all, passed, failed);
+        }
+
+        List<Scenario> all() {
+            return all;
+        }
+
+        Diff compare(Scenarios other) {
```

Comment:
I do like the acceptance tests, but unit tests for the methods `compareTo` and `verify` would be nice.

Reply:
Good point. Done.